File: web/src/entity/student.ts
/**
 * 学生
 */
export class Student {
  id: number | undefined;
  name: string | undefined;
  sno: string | undefined;
  constructor(data = {} as {
    id?: number
    name?: string,
    sno?: string,
  } as Student) {
    this.id = data.id;
    this.name = data.name;
    this.sno = data.sno;
  }
}

File: api/src/main/java/hepan/page/api/entity/Student.java
package hepan.page.api.entity;

import jakarta.persistence.*;

@Entity
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String sno;

    public Long getId() { return id; }

    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }

    public void setName(String name) { this.name = name; }

    public String getSno() { return sno; }

    public void setSno(String sno) { this.sno = sno; }
}